### PR TITLE
Support TensorFlow pluggable devices

### PR DESCRIPTION
**Pull request by jianyizh**
[PluggableDevice](https://github.com/tensorflow/community/blob/master/rfcs/20200624-pluggable-device-for-tensorflow.md) architecture offers a plugin mechanism for registering devices with TensorFlow without the need to make changes in TensorFlow code. It relies on C APIs to communicate with the TensorFlow binary in a stable manner. [Issue 2038](https://github.com/tensorflow/serving/issues/2038) also asks for support of pluggable devices.

This PR adds the support for pluggable device through a new build config `--config=plugins`. It will try to load all .so files under the given directory from argument `--tensorflow_plugins` by `TF_LoadPluggableDeviceLibrary`.
We will need to add `alwaylink=1` in TensorFlow `kernels_experimental` library.

### CODE REVIEW

File: tensorflow_serving/model_servers/BUILD [hunk 1]
```diff
@@ -383,6 +384,7 @@ cc_library(
         "server.cc",
     ],
     hdrs = ["server.h"],
+    textual_hdrs = ["tf_c_api_exported_symbols.lds"],
```
Comment (Zantares) [inline tensorflow_serving/model_servers/BUILD]: This line can be deleted if don't use generating script.

**Comment by Zantares** [comment]
With current exported symbol "*TF_*", the package size is increased 4M and total ratio is increased to 101%.

File: tensorflow_serving/model_servers/BUILD (hunk 1 above)
Comment (jianyizh) [reply tensorflow_serving/model_servers/BUILD]: deleted

**Comment by jianyizh** [comment]
open another pr to tf serving

**Comment by jzhoulon** [comment]
LGTM

**Pull request closed without merging**